tests/console/osc.pm

```diff
@@ -0,0 +1,81 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+# Author: Gustavo Figueira <gfigueira@suse.com>
+
+use base "consoletest";
+use testapi;
+use utils;
+use Utils::Backends 'use_ssh_serial_console';
+use version_utils qw(is_sle get_os_release is_tumbleweed);
+use strict;
+use warnings;
+
+sub repos_and_pkgs_sle {
+    my ($hub_version, $arch) = @_;
+    my $repo_version = $hub_version;
+    $repo_version =~ s/\./_SP/;
+    assert_script_run("SUSEConnect --product PackageHub/" . $hub_version . "/" . $arch);
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/CA/SLE_" . $repo_version . "/ suseca");
+    zypper_call("in osc obs-service-source_validator ca-certificates-suse");
+}
+
+sub prepare_oscrc {
+    my ($oscrc_path, $IBSUSER, $SECRET) = @_;
+    assert_script_run("wget --quiet " . data_url('oscrc') . " -O " . $oscrc_path, timeout => 30);
+    assert_script_run("perl -pi -e 's/TAGIBSUSER/" . $IBSUSER . "/g' " . $oscrc_path);
+    assert_script_run("perl -pi -e 's/TAGSECRET/" . $SECRET . "/g' " . $oscrc_path);
+}
+
+sub co_and_build {
+    my ($codestream, $testpkg) = @_;
+    my $api_opensuse = "https://api.opensuse.org";
+
+    if (is_sle) {
+        assert_script_run("yes 1 | osc co " . $codestream . " " . $testpkg);
+    } else {
+        assert_script_run("yes 1 | osc -A " . $api_opensuse . " co " .  $codestream . " " . $testpkg);
+    }
+    assert_script_run("cd " . $codestream . "/" . $testpkg . "; yes 1 | osc build", timeout => 300);
+}
+
+sub run {
+    check_var("BACKEND", "ipmi") ? use_ssh_serial_console : select_console 'root-console';
```

Comment:
I suggest to not distinguish code flows based on backends within test modules. This makes test code harder to read. Backend specific behaviour should be abstracted away. Likely you can use "select_serial_terminal` from lib/opensusebasetest.pm

Reply:
@ge0r -> I need your pointers here

Reply:
This is a valid point, as Oliver says `select_serial_terminal` selects which console to use based on the `BACKEND` var, so you could instead 
```suggestion
    my ($self) = @_; 
    $self->select_serial_terminal;
```